src/python/pants/util/meta.py

```diff
@@ -107,3 +109,32 @@ def staticproperty(func: Union[staticmethod, Callable]) -> ClassPropertyDescript
     func = staticmethod(func)
 
   return ClassPropertyDescriptor(func, doc)
+
+
+def freeze_after_init(cls: Type[T]) -> Type[T]:
```

Comment:
Minor naming nit: Perhaps this should be called `frozen_after_init`, as AFAICT annotations tend to be descriptive, not imperative.

Reply:
Oh I like this even more! Since this is a major API we will be using and possibly making a public API, worth getting this right.

Thanks!